modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -362,6 +358,27 @@ private void _addOrUpdateObjectFieldSettings(
 		objectFieldBusinessType.validateObjectFieldSettings(
 			objectField.getName(), objectFieldSettings);
 
+		List<ObjectFieldSetting> oldObjectFieldSettings =
+			_objectFieldSettingPersistence.findByObjectFieldId(
+				objectField.getObjectFieldId());
+
+		for (ObjectFieldSetting oldObjectFieldSetting :
+				oldObjectFieldSettings) {
+
+			Stream<ObjectFieldSetting> stream = objectFieldSettings.stream();
+
+			Optional<ObjectFieldSetting> objectFieldSettingOptional =
+				stream.filter(
+					newObjectFieldSetting -> Objects.equals(
+						newObjectFieldSetting.getName(),
+						oldObjectFieldSetting.getName())
+				).findFirst();
+
+			if (!objectFieldSettingOptional.isPresent()) {
+				_objectFieldSettingPersistence.remove(oldObjectFieldSetting);
+			}
+		}
+
```

Comment:
Hi @rodrigopaulino, just to let you know, initially I also used `stream` but Brian changed it in this [commit](https://github.com/liferay/liferay-portal/commit/f64db3aa56d9488e22611b50b955d23a6dad992b).

But I would send it this way since the `ci:test:relevant` has passed.